src/core/diags/factory_diags.cpp

```diff
@@ -570,14 +573,11 @@ Error Diags::TransmitPacket(void)
         }
     }
 
-    mDiagSendOn = true;
-    error       = Get<Radio>().Transmit(*static_cast<Mac::TxFrame *>(mTxPacket));
-
-    if (error == kErrorInvalidState)
-    {
-        mStats.mSentErrorInvalidStatePackets++;
-    }
+    error = Get<Radio>().Transmit(*static_cast<Mac::TxFrame *>(mTxPacket));
+    VerifyOrExit(error != kErrorNone, mDiagSendOn = true);
+    UpdateTxStats(error);
```

Comment:
While it is correct, it is harder to read and follow the negative boolean checks with `VerifyOrExit`:
- We want to verify that error is not `kErrorNone`, and if it is not not none (i.e., it is successful), we want to set `mDiagSendOn = true` and exit.

How about this:
```
    if (error == kErrorNone)
    {
         mDiagSendOn = true;
    }
    else
    {
         UpdateTxStats(error);
    }
```